[2023-10-11T19:27:33.224+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-11T19:27:33.233+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-11T19:27:33.233+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:27:33.250+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-05 00:00:00+00:00
[2023-10-11T19:27:33.257+0000] {standard_task_runner.py:57} INFO - Started process 464 to run task
[2023-10-11T19:27:33.262+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-05T00:00:00+00:00', '--job-id', '18', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmp0_xxxcv4']
[2023-10-11T19:27:33.265+0000] {standard_task_runner.py:85} INFO - Job 18: Subtask trusted
[2023-10-11T19:27:33.323+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:27:33.439+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-05T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-05T00:00:00+00:00'
[2023-10-11T19:28:38.889+0000] {local_task_job_runner.py:294} WARNING - State of this instance has been externally set to None. Terminating instance.
[2023-10-11T19:28:38.891+0000] {process_utils.py:131} INFO - Sending 15 to group 464. PIDs of all processes in the group: [469, 464]
[2023-10-11T19:28:38.892+0000] {process_utils.py:86} INFO - Sending the signal 15 to group 464
[2023-10-11T19:28:38.892+0000] {taskinstance.py:1630} ERROR - Received SIGTERM. Terminating subprocesses.
[2023-10-11T19:28:38.905+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/wikipedia_etl.py", line 133, in trusted
    sc = SparkContext(conf=conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/java_gateway.py", line 104, in launch_gateway
    time.sleep(0.1)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1632, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2023-10-11T19:28:38.909+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231005T000000, start_date=20231011T192733, end_date=20231011T192838
[2023-10-11T19:28:38.924+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 18 for task trusted ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(wikipedia_etl, trusted, scheduled__2023-10-05T00:00:00+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'trusted', 'dag_id': 'wikipedia_etl', 'run_id': 'scheduled__2023-10-05T00:00:00+00:00', 'map_index': -1, 'start_date': datetime.datetime(2023, 10, 11, 19, 27, 33, 225100, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2023, 10, 11, 19, 28, 38, 909181, tzinfo=Timezone('UTC')), 'duration': 65}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 464)
[2023-10-11T19:28:38.944+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=464, status='terminated', exitcode=1, started='19:27:32') (464) terminated with exit code 1
[2023-10-11T19:28:39.038+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=469, status='terminated', started='19:27:32') (469) terminated with exit code None
[2023-10-11T19:33:12.235+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-11T19:33:12.242+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-11T19:33:12.242+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:33:12.252+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-05 00:00:00+00:00
[2023-10-11T19:33:12.257+0000] {standard_task_runner.py:57} INFO - Started process 1925 to run task
[2023-10-11T19:33:12.259+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-05T00:00:00+00:00', '--job-id', '32', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpyv2ux4jm']
[2023-10-11T19:33:12.262+0000] {standard_task_runner.py:85} INFO - Job 32: Subtask trusted
[2023-10-11T19:33:12.299+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:33:12.357+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-05T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-05T00:00:00+00:00'
[2023-10-11T19:33:38.019+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-11T19:33:38.041+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231005T000000, start_date=20231011T193312, end_date=20231011T193338
[2023-10-11T19:33:38.086+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-11T19:33:38.098+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-11T19:48:39.515+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-11T19:48:39.523+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-11T19:48:39.523+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:48:39.533+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-05 00:00:00+00:00
[2023-10-11T19:48:39.538+0000] {standard_task_runner.py:57} INFO - Started process 4624 to run task
[2023-10-11T19:48:39.541+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-05T00:00:00+00:00', '--job-id', '52', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpzci1jzc6']
[2023-10-11T19:48:39.543+0000] {standard_task_runner.py:85} INFO - Job 52: Subtask trusted
[2023-10-11T19:48:39.582+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:48:39.638+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-05T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-05T00:00:00+00:00'
[2023-10-11T19:49:12.204+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-11T19:49:12.214+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231005T000000, start_date=20231011T194839, end_date=20231011T194912
[2023-10-11T19:49:12.272+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-11T19:49:12.284+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-13T17:19:55.455+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-13T17:19:55.462+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-13T17:19:55.462+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-13T17:19:55.472+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-05 00:00:00+00:00
[2023-10-13T17:19:55.478+0000] {standard_task_runner.py:57} INFO - Started process 1101 to run task
[2023-10-13T17:19:55.481+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-05T00:00:00+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmp5q2b2xi7']
[2023-10-13T17:19:55.483+0000] {standard_task_runner.py:85} INFO - Job 12: Subtask trusted
[2023-10-13T17:19:55.520+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [running]> on host feba906f3341
[2023-10-13T17:19:55.580+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-05T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-05T00:00:00+00:00'
[2023-10-13T17:20:34.825+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-13T17:20:34.838+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231005T000000, start_date=20231013T171955, end_date=20231013T172034
[2023-10-13T17:20:34.874+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-13T17:20:34.889+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-16T11:27:24.869+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-16T11:27:24.879+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-16T11:27:24.880+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-16T11:27:24.898+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-05 00:00:00+00:00
[2023-10-16T11:27:24.905+0000] {standard_task_runner.py:57} INFO - Started process 1068 to run task
[2023-10-16T11:27:24.909+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-05T00:00:00+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpa9gq3_tm']
[2023-10-16T11:27:24.911+0000] {standard_task_runner.py:85} INFO - Job 12: Subtask trusted
[2023-10-16T11:27:24.954+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [running]> on host ab44499fca46
[2023-10-16T11:27:25.040+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-05T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-05T00:00:00+00:00'
[2023-10-16T11:27:59.293+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-16T11:27:59.311+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231005T000000, start_date=20231016T112724, end_date=20231016T112759
[2023-10-16T11:27:59.353+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-16T11:27:59.374+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-16T19:10:36.892+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-16T19:10:36.899+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [queued]>
[2023-10-16T19:10:36.899+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-16T19:10:36.910+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-05 00:00:00+00:00
[2023-10-16T19:10:36.916+0000] {standard_task_runner.py:57} INFO - Started process 1055 to run task
[2023-10-16T19:10:36.919+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-05T00:00:00+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpibp34bmt']
[2023-10-16T19:10:36.921+0000] {standard_task_runner.py:85} INFO - Job 12: Subtask trusted
[2023-10-16T19:10:36.956+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-05T00:00:00+00:00 [running]> on host 3522330161cc
[2023-10-16T19:10:37.013+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-05T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-05T00:00:00+00:00'
[2023-10-16T19:11:11.294+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-16T19:11:11.305+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231005T000000, start_date=20231016T191036, end_date=20231016T191111
[2023-10-16T19:11:11.328+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-16T19:11:11.340+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
